test/simulator/test/org/apache/cassandra/simulator/test/SimulationTestBase.java

```diff
@@ -215,12 +215,16 @@ public static void simulate(IIsolatedExecutor.SerializableRunnable[] runnables,
         IsolatedExecutor.transferAdhoc((IIsolatedExecutor.SerializableConsumer<ExecutorFactory>) ExecutorFactory.Global::unsafeSet, classLoader)
                         .accept(factory);
 
+        IntSupplier intSupplier = () -> {
+            if (InterceptibleThread.isDeterministic())
+                throw failWithOOM();
+            return random.uniform(Integer.MIN_VALUE, Integer.MAX_VALUE);
+        };
+
         IsolatedExecutor.transferAdhoc((IIsolatedExecutor.SerializableBiConsumer<InterceptorOfGlobalMethods, IntSupplier>) InterceptorOfGlobalMethods.Global::unsafeSet, classLoader)
-                        .accept(interceptorOfGlobalMethods, () -> {
-                            if (InterceptibleThread.isDeterministic())
-                                throw failWithOOM();
-                            return random.uniform(Integer.MIN_VALUE, Integer.MAX_VALUE);
-                        });
+                        .accept(interceptorOfGlobalMethods, intSupplier);
+
+        InterceptorOfGlobalMethods.Global.unsafeSet(interceptorOfGlobalMethods, intSupplier);
```

Comment:
@belliottsmith I believe you said that this is not the correct behavior and unsafe... This was needed to avoid ClassLoader issues, not sure if/how to fix the test to avoid